examples/pub-sub-rewind/javascript/src/script.ts

```diff
@@ -0,0 +1,184 @@
+    const numMarketsToUpdate = Math.floor(Math.random() * 2) + 1;
+    const marketsToUpdate = markets.sort(() => 0.5 - Math.random()).slice(0, numMarketsToUpdate);
+
+    marketsToUpdate.forEach((market) => {
+      matchData.match.matchOdds[market] = (parseFloat(matchData.match.matchOdds[market]) + (Math.random() * 0.2 - 0.1)).toFixed(2);
+    });
+
+    matchData.match.timestamp = new Date().toISOString();
+    await channel.publish('odds', matchData);
+
+    // Show alert for each publish
+    const alert = document.createElement('div');
+    alert.className =
+      'fixed bottom-4 right-4 bg-green-500 text-white px-4 py-2 rounded shadow-lg transition-opacity duration-500';
+    alert.textContent = `Update ${i + 1}/10: New odds published`;
+    document.body.appendChild(alert);
+
+    // Remove alert after 2 seconds
+    setTimeout(() => {
+      alert.style.opacity = '0';
+      setTimeout(() => alert.remove(), 500);
+    }, 2000);
+
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+  }
+
+  await enterGame();
+});
+
+async function updateCurrentOdds(message: Message) {
+  const score = document.getElementById('score');
+  score.textContent = message.data.match.score;
+  const currentHome = document.getElementById('current-home');
+  currentHome.textContent = message.data.match.matchOdds.homeWin;
+  const currentAway = document.getElementById('current-away');
+  currentAway.textContent = message.data.match.matchOdds.awayWin;
+  const currentDraw = document.getElementById('current-draw');
+  currentDraw.textContent = message.data.match.matchOdds.draw;
+}
+
+async function addHistoryItem(message: Message, position = 'prepend') {
+  const history = document.getElementById('history');
+  const historyItem = document.createElement('div');
+  historyItem.id = `history-item-${message.id}`;
+  historyItem.className = 'border-b pb-2';
+  const historyDiv = document.createElement('div');
+  historyDiv.className = 'flex justify-between text-sm text-gray-600';
+  historyItem.appendChild(historyDiv);
+
+  const homeWin = document.createElement('span');
+  homeWin.textContent = `Home: ${message.data.match.matchOdds.homeWin}`;
+  const draw = document.createElement('span');
+  draw.textContent = `Draw: ${message.data.match.matchOdds.draw}`;
+  const awayWin = document.createElement('span');
+  awayWin.textContent = `Away: ${message.data.match.matchOdds.awayWin}`;
+  const time = document.createElement('span');
+  time.textContent = new Date(message.data.match.timestamp).toLocaleTimeString([], {
+    hour: '2-digit',
+    minute: '2-digit',
+    second: '2-digit',
+  });
+  historyDiv.appendChild(homeWin);
+  historyDiv.appendChild(draw);
+  historyDiv.appendChild(awayWin);
+  historyDiv.appendChild(time);
+
+  if (position === 'prepend') {
+    history.prepend(historyItem);
+  } else {
+    history.appendChild(historyItem);
+  }
+}
+
+async function updateRandomOdds() {
+  if (!matchData) {
+    return;
+  }
+
+  let running = true;
```

Comment:
Could you explain why we need `running`?

Reply:
It was there to make sure that the loop in `updateRandomOdds function` is stopped if the user navigates away from the page, but I think it's no longer needed. I've removed it.